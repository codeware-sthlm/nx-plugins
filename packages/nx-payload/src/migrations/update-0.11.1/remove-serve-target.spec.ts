import {
  type Tree,
  addProjectConfiguration,
  readNxJson,
  updateNxJson
} from '@nx/devkit';
import { createTreeWithEmptyWorkspace } from '@nx/devkit/testing';

import removeServeTarget from './remove-serve-target';

function addProject(
  tree: Tree,
  name: string,
  type: 'payload-app' | 'normal-app'
) {
  if (type === 'payload-app') {
    tree.write(`apps/${name}/payload.config.ts`, '');
  }
  addProjectConfiguration(tree, name, {
    name,
    root: `apps/${name}`,
    sourceRoot: `apps/${name}/src`,
    projectType: 'application',
    targets: {
      build: {
        executor: '@nx/js:tsc',
        options: {}
      },
      serve: {
        executor: '@nx/js:node',
        options: {}
      }
    }
  });
}

function addPayloadPlugin(tree: Tree) {
  const nxJson = readNxJson(tree);
  nxJson.plugins ??= [];
  nxJson.plugins.push('@cdwr/nx-payload/plugin');
  updateNxJson(tree, nxJson);
}

describe('remove-serve-target migration', () => {
  let tree: Tree;

  console.log = jest.fn();
  console.warn = jest.fn();

  beforeEach(() => {
    tree = createTreeWithEmptyWorkspace({ layout: 'apps-libs' });
  });

  it('should remove serve target from payload projects', async () => {
    addProject(tree, 'test-payload', 'payload-app');
    addProject(tree, 'test-normal', 'normal-app');
    addPayloadPlugin(tree);

    await removeServeTarget(tree);

    expect(
      tree.read('apps/test-payload/project.json', 'utf-8')
    ).toMatchSnapshot('payload project');

    expect(tree.read('apps/test-normal/project.json', 'utf-8')).toMatchSnapshot(
      'normal project'
    );
  });

  it('should not remove serve targets when payload plugin is not installed', async () => {
    addProject(tree, 'test-payload', 'payload-app');
    addProject(tree, 'test-normal', 'normal-app');

    await removeServeTarget(tree);

    expect(
      tree.read('apps/test-payload/project.json', 'utf-8')
    ).toMatchSnapshot('payload project');

    expect(tree.read('apps/test-normal/project.json', 'utf-8')).toMatchSnapshot(
      'normal project'
    );
  });
});
