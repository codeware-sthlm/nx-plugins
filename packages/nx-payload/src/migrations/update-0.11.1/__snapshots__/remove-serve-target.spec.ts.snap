// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`remove-serve-target migration should not remove serve targets when payload plugin is not installed: normal project 1`] = `
"{
  "name": "test-normal",
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "apps/test-normal/src",
  "projectType": "application",
  "targets": {
    "build": {
      "executor": "@nx/js:tsc",
      "options": {}
    },
    "serve": {
      "executor": "@nx/js:node",
      "options": {}
    }
  }
}
"
`;

exports[`remove-serve-target migration should not remove serve targets when payload plugin is not installed: payload project 1`] = `
"{
  "name": "test-payload",
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "apps/test-payload/src",
  "projectType": "application",
  "targets": {
    "build": {
      "executor": "@nx/js:tsc",
      "options": {}
    },
    "serve": {
      "executor": "@nx/js:node",
      "options": {}
    }
  }
}
"
`;

exports[`remove-serve-target migration should remove serve target from payload projects: normal project 1`] = `
"{
  "name": "test-normal",
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "apps/test-normal/src",
  "projectType": "application",
  "targets": {
    "build": {
      "executor": "@nx/js:tsc",
      "options": {}
    },
    "serve": {
      "executor": "@nx/js:node",
      "options": {}
    }
  }
}
"
`;

exports[`remove-serve-target migration should remove serve target from payload projects: payload project 1`] = `
"{
  "name": "test-payload",
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "apps/test-payload/src",
  "projectType": "application",
  "targets": {
    "build": {
      "executor": "@nx/js:tsc",
      "options": {}
    }
  }
}
"
`;
